src/pages/home/components/categories/Categories.tsx

```diff
@@ -1,11 +1,12 @@
-import React from "react";
+import React, {useState} from "react";
 import Container from "@mui/material/Container";
 import Typography from "@mui/material/Typography";
-
 import Box from "@mui/material/Box";
+import { RulesModal } from "../../../../components/index";
 import { categoriesData } from "./categoriesData";
 
 export const CategoriesContainer = () => {
+    const [openRulesModal, setOpenRulesModal] = useState<any>(false);
```

Comment:
since you have explicitly mentioned that the state variable is going to be a boolean value so you don't need to provide type event thought if you provide it so don't add as any

this can be refactored to as
const [openRulesModal, setOpenRulesModal] = useState<boolean>(false);

🙂😀

Reply:
this is material ui issue so I have to provide type event any